modules/core/src/test/java/org/apache/ignite/logger/java/JavaLoggerFormatterTest.java

```diff
@@ -0,0 +1,178 @@
+     */
+    @Test
+    public void testNullCategory() {
+        assertFormattedOutput(null, JavaLoggerFormatter.ANONYMOUS_LOGGER_NAME);
+    }
+
+    /**
+     * Empty (zero-length) category should fall back to anonymous
+     */
+    @Test
+    public void testEmptyCategory() {
+        assertFormattedOutput("", JavaLoggerFormatter.ANONYMOUS_LOGGER_NAME);
+    }
+
+    /**
+     * Should use specified literal as category
+     */
+    @Test
+    public void testSimpleCategory() {
+        assertFormattedOutput("foo", "foo");
+    }
+
+    /**
+     * Should use classname as category
+     */
+    @Test
+    public void testClassCategory() {
+        assertFormattedOutput(JavaLoggerFormatterTest.class.getName(), "JavaLoggerFormatterTest");
+    }
+
+    /**
+     * Should append stack trace to log message
+     */
+    @Test
+    public void testThrowable() {
+        // given
+        final String msg = "Greed is good";
+        final String category = "probe";
+        final String exceptionMsg = "Not enough minerals";
+
+        LogRecord record = buildLogRecord(msg, category);
+        try {
+            throw new RuntimeException(exceptionMsg);
+        }
+        catch (Throwable t) {
+            record.setThrown(t);
+        }
+
+        // when
+        String result = formatter.format(record);
+
+        // then
+        final String expected = buildExpectedString(msg, category) + "\njava.lang.RuntimeException: " + exceptionMsg;
+        assertSimilar(expected, result);
+    }
+
+    /**
+     * Should use date format from property
+     */
+    @Test
+    public void testCorrectCustomFormatProperty() {
+        // given
+        System.setProperty("IGNITE_JAVA_LOGGER_DATE_FORMAT", "yyyy/MM/dd HH:mm:ss");
+
+        // when
+        DateTimeFormatter fmt = JavaLoggerFormatter.resolveDateFormat();
+
+        // then
+        assertEquals("2000/01/01 01:01:01", fmt.format(Instant.ofEpochMilli(BASE_MILLIS)));
+    }
+
+    /**
+     * Should use default when date format from property is invalid
+     */
+    @Test
+    public void testIncorrectCustomFormatProperty() {
+        // given
+        System.setProperty("IGNITE_JAVA_LOGGER_DATE_FORMAT", "foobar");
+
+        // when
+        DateTimeFormatter fmt = JavaLoggerFormatter.resolveDateFormat();
+
+        // then
+        assertEquals(BASE_DATE, fmt.format(Instant.ofEpochMilli(BASE_MILLIS)));
+    }
+
+    void assertFormattedOutput(String msg, String category, String expectedCategory) {
+        // given
+        LogRecord record = buildLogRecord(msg, category);
+
+        // when
+        String result = formatter.format(record);
+
+        // then
+        assertSimilar(buildExpectedString(msg, expectedCategory), result);
+    }
+
+    void assertFormattedOutput(String category, String expectedCategory) {
+        assertFormattedOutput("Power overwhelming", category, expectedCategory);
+    }
+
+    static final DateTimeFormatter ISO_DATE_FORMAT =
```

Comment:
What is this variable for?

Reply:
This formatter was created for composing expected date format. I defined it to guard us from occasional change of date format inside JavaLoggerFormatter (e.g. you can't use code under test for testing this code). But seems, indeed, usage of IgniteUtils.ISO_DATE_FMT should be safe enough